### ISSUE TITLE
Error while downloading elm-json

### ISSUE THREAD

**Issue by wolfadex** [jfmengels/node-elm-review#87]
After running `elm-review --fix-all --config review/zamboni` in our CI, I see the following

```
Downloading elm-json... 0%
Downloading elm-json... 0%
Downloading elm-json... 22%
Downloading elm-json... 57%
-- UNEXPECTED ERROR ------------------------------------------------------------

I ran into an unexpected error. Please open an issue at the following link:
  https://github.com/jfmengels/node-elm-review/issues/new

Please include this error message and as much detail as you can provide. If you
can, please provide a setup that makes it easy to reproduce the error. That will
make it much easier to fix the issue.

Below is the error that was encountered.
--------------------------------------------------------------------------------
TypeError: process.stdout.clearLine is not a function
    at Object.clearAndLog (/home/runner/work/ironzion/ironzion/node_modules/elm-review/lib/styled-message.js:12:20)
    at Array.<anonymous> (/home/runner/work/ironzion/ironzion/node_modules/elm-review/lib/autofix.js:25:21)
    at Function.f (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:2386:19)
    at A3 (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:59:28)
    at Object.b (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:2145:7)
    at _Scheduler_step (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:1989:20)
    at _Scheduler_enqueue (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:1951:3)
    at /home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:1979:5
    at Timeout._onTimeout (/home/runner/work/ironzion/ironzion/packages/frontend/elm-stuff/generated-code/jfmengels/elm-review/cli/2.8.2/review-applications/ffb26063051ea63e1df365f8af361ed6.js:2009:4)
    at listOnTimeout (node:internal/timers:559:[17](https://github.com/Blissfully/ironzion/actions/runs/3555940445/jobs/5972918260#step:5:18))
```

This has happened 3 times (I only became aware today as it only runs on weekends) and each time it's reached a different percent, anywhere from 1% to 81%.

The offending line looks to be https://github.com/jfmengels/node-elm-review/blob/8aa63d0b6374a084b3ff8ccc53638b0be07e90c3/lib/styled-message.js, though I don't think this is the root cause of the failure, but a secondary issue. An initial run of elm-review locally isn't showing anything but I'm going to keep testing to see if I can find the cause.

**Comment by lydell** [comment]
I can share some knowledge here.

`.clearLine` is only defined if `.isTTY` is defined. (Interestingly, TypeScript does not capture this fact.)

```
❯ node -p '[process.stdout.isTTY, process.stdout.clearLine]'
[ true, [Function (anonymous)] ]

❯ node -p '[process.stdout.isTTY, process.stdout.clearLine]' | cat
[ undefined, undefined ]
```

CI platforms typically aren’t TTYs: Many support colors, but not cursor movements. Judging by the output you pasted, cursor movements are _not_ supported in your CI, since the `Downloading elm-json... X%` messages form a log, rather than updating the same line all the time.

So it looks like there might be some isTTY check missing.

**Comment by jfmengels** [comment]
Thanks for the issue and for the insight @lydell, that was very helpful.

I have released the fix in v`2.8.3`. Hopefully that will solve the issue :slightly_smiling_face:
